File: ruby/quick_sort.rb
def quick_sort(arr)
  return arr if arr.length <= 1

  pivot = arr[arr.length / 2]
  left = arr.select { |x| x < pivot }
  middle = arr.select { |x| x == pivot }
  right = arr.select { |x| x > pivot }

  quick_sort(left) + middle + quick_sort(right)
end

# Example usage
arr = [64, 34, 25, 12, 22, 11, 90]
sorted_arr = quick_sort(arr)
puts "Sorted array: #{sorted_arr}"

File: php/quick_sort.php
<?php

function quickSort($arr) {
    if (count($arr) < 2) return $arr;
    
    $left = $right = [];
    $pivot = array_shift($arr);
    
    foreach ($arr as $v) {
        if ($v < $pivot) $left[] = $v;
        else $right[] = $v;
    }
    
    return array_merge(quickSort($left), [$pivot], quickSort($right));
}

// Example usage
$arr = [64, 34, 25, 12, 22, 11, 90];
$sortedArr = quickSort($arr);
echo "Sorted array: ";
print_r($sortedArr);

?>
